omnisend-for-formidable-forms-add-on/includes/Factory/class-omnisendresponsefactory.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Omnisend response factory
+ *
+ * @package OmnisendFormidableFormsPlugin
+ */
+
+declare(strict_types=1);
+
+namespace Omnisend\FormidableFormsAddon\Factory;
+
+use Omnisend\FormidableFormsAddon\OmnisendResponse;
+
+/**
+ * Omnisend response factory.
+ */
+class OmnisendResponseFactory {
+
+	/**
+	 * Create an Omnisend response.
+	 *
+	 * @param bool        $success Whether the response is successful.
+	 * @param string      $email The email associated with the response.
+	 * @param string|null $phone The phone number associated with the response.
+	 *
+	 * @return OmnisendResponse The created Omnisend response.
+	 */
+	public function create( bool $success, string $email = '', ?string $phone = '' ): OmnisendResponse {
```

Comment:
I guess code would be more simpler if you move (and make static) this function to OmnisendResponse

Reply:
Static functions are not considered to be best practise, trying to avoid using them.

Reply:
I guess it is bad in some cases, for example when you depend on global state, mutate static variables, etc, but here it is just simple object creation without any dependency, so it looks logical keep related logic together. Anyway, probably there is no clear good or bad. Resolving this.